application/common/models/User.php

```diff
@@ -92,6 +92,12 @@ public function beforeDelete()
         }
 
         foreach ($this->mfas as $mfa) {
+            // first delete the children and then the parent to avoid foreign key constraint error
+            if ($mfa->type == MFA::TYPE_WEBAUTHN) {
+                if (!self::deleteWebauthns($mfa)) {
```

Comment:
It will probably be cleaner to leave this `User->beforeDelete()` method unchanged, and instead update the Mfa record's `beforeDelete()` method to delete each of its MfaWebauthn children records when appropriate (using each one's `delete()` method).